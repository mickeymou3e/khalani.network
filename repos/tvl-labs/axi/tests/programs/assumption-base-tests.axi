const a: Boolean;
const b: Boolean;

# Test duplicated assume.
assume a {
    assume a {
        !claim(a)
    }
    !right_either(b, a)
}

# Test that b doesn't hold when the assume b block is closed.
assume a {
    assume b {
        !both(a, b)
    }
    !claim(a & b)
}
