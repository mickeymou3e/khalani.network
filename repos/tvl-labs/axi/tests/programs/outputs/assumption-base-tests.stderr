[31mError:[0m 
    [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m<unknown>:17:12[38;5;246m][0m
    [38;5;246mâ”‚[0m
 [38;5;246m17 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m![0m[38;5;249mc[0m[38;5;249ml[0m[38;5;249ma[0m[38;5;249mi[0m[38;5;249mm[0m[38;5;249m([0ma & b[38;5;249m)[0m
 [38;5;240m   â”‚[0m            â”€â”€â”¬â”€â”€  
 [38;5;240m   â”‚[0m              â•°â”€â”€â”€â”€ proposition does not hold
[38;5;246mâ”€â”€â”€â”€â•¯[0m
